core/controllers/reader_test.py

```diff
@@ -362,3 +363,166 @@ def test_ratings_by_different_users(self):
+            'newuser has submitted a new report on the exploration Title'
+            ' on the grounds of AD .\n'
+            'You can modify the exploration by clicking "Edit Title".\n'
+            '\n'
+            'Thanks!\n'
+            '- The Oppia Team\n'
+            '\n'
+            'You can change your email preferences via the Preferences page.')
+
+        with self.can_send_emails_ctx:
+
+            self.process_and_flush_pending_tasks()
+
+            messages = self.mail_stub.get_sent_messages(to=self.MODERATOR_EMAIL)
+            self.assertEqual(len(messages), 1)
+            self.assertEqual(
+                messages[0].html.decode(),
+                expected_email_html_body)
+            self.assertEqual(
+                messages[0].body.decode(),
+                expected_email_text_body)"""
+
+
+class FlagExplorationEmailHandlerTest(test_utils.GenericTestBase):
+
+    """def setUp(self):
+        super(FlagExplorationEmailHandlerTest, self).setUp()
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+
+        self.signup(self.NEW_USER_EMAIL, self.NEW_USER_USERNAME)
+        self.new_user_id = self.get_user_id_from_email(self.NEW_USER_EMAIL)
+
+        self.signup(self.MODERATOR_EMAIL, self.MODERATOR_USERNAME)
+        self.moderator_id = self.get_user_id_from_email(self.MODERATOR_EMAIL)
+        self.set_moderators([self.MODERATOR_USERNAME])
+
+        self.exploration = self.save_new_default_exploration(
+            'A', self.editor_id, 'Title')
+        self.can_send_emails_ctx = self.swap(
+            feconf, 'CAN_SEND_EMAILS', True)
+        self.report_text = 'AD'
+
+    def test_that_emails_are_sent(self):
+        expected_email_html_body = (
+            'Hello Moderator,<br>'
+            'newuser has submitted a new report on the exploration Title'
+            ' on the grounds of AD .<br>'
+            'You can modify the exploration by clicking '
+            '<a href="https://www.oppia.org/create/A">'
+            '"Edit Title"</a>.<br>'
+            '<br>'
+            'Thanks!<br>'
+            '- The Oppia Team<br>'
+            '<br>'
+            'You can change your email preferences via the '
+            '<a href="https://www.example.com">Preferences</a> page.')
+
+        expected_email_text_body = (
+            'Hello Moderator,\n'
+            'newuser has submitted a new report on the exploration Title'
+            ' on the grounds of AD .\n'
+            'You can modify the exploration by clicking "Edit Title".\n'
+            '\n'
+            'Thanks!\n'
+            '- The Oppia Team\n'
+            '\n'
+            'You can change your email preferences via the Preferences page.')
+
+        with self.can_send_emails_ctx:
+            moderator_services.enqueue_flag_exploration_email_task(
+                self.exploration.id, self.report_text)
+
+            self.process_and_flush_pending_tasks()
```

Comment:
Firstly, thanks for explaining what you did! That makes it a lot easier to help.

That said, in this case, I actually don't know what the issue is, though I doubt it's anything to do with ALLOWED_HOST if the other similar tests pass. 500 is a server error that indicates that something's going wrong somewhere, server-side. Maybe an exception is being thrown, or an invalid key is being referred to, or something.

The problem here is that you're not getting good enough debug information. If you put a `raise Exception('a')` at the very beginning of that handler, do you get the same error?

If so, this means you need to find a way to observe the 'a' in Exception('a'). If you run Oppia locally, are you able to do that? If not, see if you can figure out how to do so...